robottelo/entities.py

```diff
@@ -587,15 +566,24 @@ def path(self, which=None):
 
         """
         if which in (
-                'content_view_puppet_modules', 'content_view_versions',
-                'publish', 'available_puppet_module_names',
-                'available_puppet_modules'):
+                'available_puppet_module_names',
+                'available_puppet_modules',
+                'content_view_puppet_modules',
+                'content_view_versions',
+                'publish'):
             return '{0}/{1}'.format(
                 super(ContentView, self).path(which='self'),
                 which
             )
         return super(ContentView, self).path(which)
 
+    def read(self, auth=None, entity=None, attrs=None, ignore=()):
+        """Compensate for the pluralization of the ``repository`` field."""
+        if attrs is None:
+            attrs = self.read_json(auth)
+        attrs['repositorys'] = attrs.pop('repositories')
```

Comment:
Yeah. Weird, right? I'll try to explain by way of example.

Let's say you have an entity like this:

class Vault():
    gpg_key = entity_fields.OneToManyField('GPGKey')

When you submit data to the server, this is what the JSON payload will probably look like:

    {'gpg_key_ids': [1, 2]}

And when you read a "vault" entity, you will likely receive this JSON data back:

    {'gpg_keys': [{'id': 1, 'key': …}, {'id': 2, 'key': …}]}

This is the usual pattern with the API. You submit a list of IDs and append `_ids` to the field name, and you get back a list of dictionaries and append `s` to the field name.

The `repository` field breaks this (already stupid) pattern. You submit a list of IDs named `repository_ids` and receive back a list of dictionaries named `repositories`. But there's code that is looking for a data from the server named `field_name + 's'`. What I'm doing is mangling this key name so that that code will work correctly.